pkg/provider/azure_loadbalancer.go

```diff
@@ -1820,31 +1821,53 @@ func (az *Cloud) reconcileFrontendIPConfigs(clusterName string, service *v1.Serv
 	lbFrontEndIPConfigNewConfigs := []network.FrontendIPConfiguration{}
 	changedTotal := false
 	var subnet network.Subnet
+
+	newConfigs := map[bool][]network.FrontendIPConfiguration{}
+	pipRG := az.getPublicIPAddressResourceGroup(service)
+	// Put all LB's FIPs into newConfigs according to IP family
+	if lb.FrontendIPConfigurations != nil {
+		for i := range *lb.FrontendIPConfigurations {
+			fip := (*lb.FrontendIPConfigurations)[i]
+			fipIsIPv6, err := az.isFIPIPv6(pipRG, &fip)
+			if err != nil {
+				return ownedFIPConfigs, toDeleteConfigs, changedTotal, err
+			}
+			newConfigs[fipIsIPv6] = append(newConfigs[fipIsIPv6], fip)
+		}
+	}
+
 	handleFrontendIPConfig := func(isIPv6 bool) error {
-		ownedFIPConfig, toDeleteConfigsSingleStack, changed, newConfigs, err := az.reconcileFrontendIPConfigsSingleStack(clusterName, service, lb, lbStatus, wantLb, isIPv6, lbFrontendIPConfigName[isIPv6], &subnet)
+		ownedFIPConfig, toDeleteConfigsSingleStack, changed, newConfigs, err := az.reconcileFrontendIPConfigsSingleStack(clusterName, service, lb, lbStatus, wantLb, isIPv6, newConfigs[isIPv6], lbFrontendIPConfigName[isIPv6], &subnet)
```

Comment:
Since we have had lb here in the parameters, why bother to build a list fip configs outside of the function?

Reply:
Because `vXEnabled` -> `handleFrontendIPConfig()` -> `reconcileFrontendIPConfigsSingleStack()`.
If the Service is single stack, we need to handle the `newConfigs` of the other IP family outside.